lib/merlin_recovery.ml

```diff
@@ -192,9 +192,28 @@ struct
     in
     { line; min_col; max_col; env }
 
+  let indentation_heuristic line col candidates =
```

Comment:
Could you add a comment to this function? I think it would be nice to add clarification comments by small steps.

Reply:
Also, I would add a return type for this function because it is very unclear from the first glance.